src/Mantasflowers.WebApi/Validation/Payment/PostCreateCouponRequestValidator.cs

```diff
@@ -0,0 +1,31 @@
+﻿using FluentValidation;
+using Mantasflowers.Contracts.Payment.Request;
+using System;
+
+namespace Mantasflowers.WebApi.Validation.Payment
+{
+    public class PostCreateCouponRequestValidator : AbstractValidator<PostCreateCouponRequest>
+    {
+        public PostCreateCouponRequestValidator()
+        {
+            RuleFor(x => x.DiscountPrice)
+                .NotEmpty()
+                .GreaterThan(0);
+
+            RuleFor(x => x.DurationInMonths)
+                .NotEmpty()
+                .GreaterThan(0);
+
+            RuleFor(x => x.Name)
+                .NotEmpty();
+
+            RuleFor(x => x.OrderOverPrice)
+                .NotEmpty()
+                .GreaterThanOrEqualTo(x => x.DiscountPrice);
```

Comment:
Related to my comment about these being decimal. If `DiscountPrice` is percent, then maybe this isn't correct?

Reply:
It is not a percent. Over the given order price deduct discount amount.